libs/api/domains/financial-statements-inao/src/lib/client/dataverseClient.ts

```diff
@@ -0,0 +1,69 @@
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import type { ConfigType } from '@island.is/nest/config'
+
+import { Inject, Injectable } from '@nestjs/common'
+
+import { dataverseClientConfig } from './dataverseClient.config'
+import { ClientType } from '../models/clientType.model'
+import { Election } from '../models/election.model'
+
+@Injectable()
+export class DataverseClient {
+  constructor(
+    @Inject(dataverseClientConfig.KEY)
+    private config: ConfigType<typeof dataverseClientConfig>,
+  ) {}
+
+  basePath = this.config.basePath
+
+  fetch = createEnhancedFetch({
+    name: 'financialStatementsInao-odata',
+    autoAuth: {
+      issuer: this.config.issuer,
+      clientId: this.config.clientId,
+      clientSecret: this.config.clientSecret,
+      scope: [this.config.scope],
+      mode: 'token',
+      tokenEndpoint: this.config.tokenEndpoint,
+    },
+  })
+
+  async getElections() {
+    const url = `${this.basePath}/new_elections`
```

Comment:
The prefix `new_` could changed based on DataVerse environment. Perhaps the prefix would be suited as a variable.

This might be best solved as a future PR perhaps when we know for sure.

Reply:
Will add in future PR